sdcm/utils/common.py

```diff
@@ -312,6 +313,18 @@ def set_public_access(self, key):
         acl_obj.put(ACL='', AccessControlPolicy={'Grants': grants, 'Owner': acl_obj.owner})
 
     def download_file(self, link, dst_dir):
+        """Download file from S3 bucket or Argus proxy link."""
+
+        if not self.s3_host_name_regex.match(urlparse(link).hostname):
+            # get the actual s3 link from Argus first
+            creds = KeyStore().get_argus_rest_credentials()
+            headers = {"Authorization": f"token {creds['token']}", **creds["extra_headers"]}
+
+            response = requests.head(link, allow_redirects=True, headers=headers)
+            link = response.history[-1].headers.get('location', link)
```

Comment:
One thing here - `Location` would be capitalized in this case I think, unless requests handles this.

Reply:
requests gets it either way (it print them lower-case)